src/semgrep_agent/meta.py

```diff
@@ -43,7 +43,9 @@ def repo(self) -> gitpython.Repo:  # type: ignore
     def repo_name(self) -> str:
         if not self.repo.head.is_valid():
             raise RuntimeError("Semgrep action cannot run on repository with no HEAD")
-        return str(os.path.basename(self.repo.working_tree_dir))
+        return os.getenv("SEMGREP_REPO_NAME") or str(
```

Comment:
If the repo name is set from an env var, we probably don't need to raise an error above, we could move the first possible return to above that if condition.

Reply:
I feel like this actually goes back to a question of do we require semgrep-action to be run on a git-controlled repo. If not, I think we can do as you say, but otherwise I still think this check is important because without a valid repo object, other things in the codebase may break. In the case of buildkite users, it seems like we still want to check for a valid repo, we just also want to let them customize name to include org_name, for example